src/Data/Text/Utf8/AhoCorasick/Automaton.hs

```diff
@@ -0,0 +1,439 @@
+-- Alfred-Margaret: Fast Aho-Corasick string searching
+-- Copyright 2022 Channable
+--
+-- Licensed under the 3-clause BSD license, see the LICENSE file in the
+-- repository root.
+
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- | UTF-8 version of 'Data.Text.AhoCorasick.Automaton'
+module Data.Text.Utf8.AhoCorasick.Automaton where
```

Comment:
I saw that you removed some comments that were present in the UTF-16 version. I think it'd be nice to include here, then we don't accidentally lose them once we remove the UTF-16 implementation.